tests/testthat/test-new-blog-post.R

```diff
@@ -0,0 +1,68 @@
+test_that("Create a blog post", {
+  skip_if_no_quarto("1.4")
+
+  current_dir <- getwd()
+
+  temp_dir <- withr::local_tempdir()
+  dir_path <- fs::path(temp_dir, "test-blog-project")
+
+  withr::defer(fs::dir_delete(dir_path), envir = rlang::current_env())
```

Comment:
`withr::local_tempdir()` already include a `defer` call to `unlink()` recursively its content. So I believe no need for a new `defer` call